src/core/execute.c

```diff
@@ -846,6 +846,12 @@ static int setup_pam(
                         goto fail;
         }
 
+        STRV_FOREACH(e, *env) {
+                pam_code = pam_putenv(handle, *e);
+                if (pam_code != PAM_SUCCESS)
+                        goto fail;
```

Comment:
You are right, using `e` is problematic, as it is assumed in the `fail:` block that `e` is `NULL`-terminated, which it is not in our case. Shall I just introduce a new variable and submit a pull request?